text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,363 @@
+# L2 Constructs for AWS VpcLattice
+
+**Status:** (DRAFT)
+
+* **Original Author(s):** @mrpackethead, 
+, @taylaand,  @nbaillie
+* **Tracking Issue:** #502
+* **API Bar Raiser:** @TheRealAmazonKendra
+
+This RFC proposes a new L2 module for CDK to support AWS VPC Lattice.
+
+---
+## PUBLIC ISSUES
+* (vpclattice): L2 for Amazon VPC Lattice #25452
+
+## Prototype Code:
+https://github.com/raindancers/latticeplay
+https://github.com/raindancers/aws-cdk/tree/mrpackethead/aws-vpclattice-alpha/packages/%40aws-cdk/aws-vpclattice-alpha
+
+---
+## VpcLattice
+
+Amazon VPC Lattice is an application networking service that consistently connects, monitors, and secures communications between your services, helping to improve productivity so that your developers can focus on building features that matter to your business. You can define policies for network traffic management, access, and monitoring to connect compute services in a simplified and consistent way across instances, containers, and serverless applications.
+
+The resources, that are required by lattice broadly fall into two catagories 
+* those are a associated with the creation of a Lattice Service network, its access and logging
+* those that are associated with the the **services** that the applicaton network will deliver. 
+
+## Example Implementation
+
+#### Create a Service Network, and allow access from a VPC
+
+
+```typescript
+import { ServiceNetwork } from '@aws-cdk/aws-vpclattice-alpha';
+
+const myServiceNetwork: ServiceNetwork;
+myServiceNetwork = new ServiceNetwork(this, 'myserviceNetwork, {
+  name: 'mylatticenetwork',
+});
+```
+Optionally an authentication policy can be added to the service network, Typically access policies for the Service Network are 
+coarse such as allowing particular accounts to use the service network. Note: the `.addLaticeAuthPolicy()` method provides defaults for 
+resources, actions and effect
+
+```typescript
+myServiceNetwork.addLatticeAuthPolicy(
+  [
+    new iam.PolicyStatement({
+      principal: new iam.AccountPrincipal('12345678900')
+    }),
+  ]
+);
+  iam.policyDocument: iam.PolicyStatement[]): void;
```

Comment:
This example doesn't seem to be formatted quite right. Can you take another look at this?

Reply:
New example provided, hopefuly the formating is correct